server/src/main/java/org/elasticsearch/rest/BaseRestHandler.java

```diff
@@ -188,6 +199,13 @@ default void close() {}
 
     public interface RequestBodyChunkConsumer extends RestChannelConsumer {
         void handleChunk(RestChannel channel, ReleasableBytesReference chunk, boolean isLast);
+
+        /**
+         * Called when the stream closes. This could happen prior to the completion of the request if the underlying channel was closed.
+         * Implementors should do their best to clean up resources and early terminate request processing if it is triggered before a
+         * response is generated.
+         */
+        default void streamClose() {}
```

Comment:
why not `close`?

Reply:
There is already a close on `RestChannelConsumer` which is triggered automatically after `prepareRequest` is called.